### DISCUSSION TITLE
How to force an SDK version for build?

### DISCUSSION THREAD

**Discussion by rectified95** [actions/virtual-environments#5359] [General questions]
I can see that the specific version of the SDK which I require (10.0.**20348**.0) is installed in the Windows2019 environment - https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#installed-windows-sdks

How can I force the right headers to be used during my build? I know it's not using the right version since my build fails with an error I resolved locally by using the proper SDK.

(Specifically, I need it to use this struct: https://docs.microsoft.com/en-us/windows/win32/api/winsock2/ns-winsock2-sock_notify_registration defined in winsock2.h)

**Comment by dsame** [comment]
Hello @rectified95,

I suppose "resolved locally by using the proper SDK" literally means you've changed target SDK of your Visual Studio project according to https://docs.microsoft.com/en-us/cpp/windows/how-to-use-the-windows-10-sdk-in-a-windows-desktop-application?view=msvc-170

Are you sure your committed the changes to the repo? I mean is not the project file accidentally gitignored? Can you please doublecheck the .cproj file refers to the correct SDK in 

```
<Project Sdk="????">

</Project>
```

**Reply by rectified95** [reply]
I should have written 'installing' rather than 'using' - the problem went away locally just after installing the SDK. I am not using VS for this, but CMAKE generators.
I ended up being able to fix the problem by passing `-DCMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION_MAXIMUM=<version>`
